bin/magpie-hive-daemon.sh

```diff
@@ -0,0 +1,208 @@
+#  under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#  
+#  Magpie is distributed in the hope that it will be useful, but
+#  WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  General Public License for more details.
+#  
+#  You should have received a copy of the GNU General Public License
+#  along with Magpie.  If not, see <http://www.gnu.org/licenses/>.
+#############################################################################
+
+# This script launches Hive for the user
+
+# First argument is path to Hive Configuration directory
+
+# Second argument is path to Hive binary
+
+# Third argument is start or stop
+
+if [ "$1X" == "X" ]
+then
+    echo "User must specify Hive configuration path"
+    exit 1
+fi
+
+if [ ! -d $1 ]
+then
+    echo "Hive configuration path is not a directory"
+fi
+
+if [ "$2X" == "X" ]
+then
+    echo "User must specify path to Hive as second argument"
+    exit 1
+fi
+
+if [ ! -d $2 ]
+then
+    echo "Hive path is not a directory"
+fi
+
+if [ "$3X" == "X" ]
+then
+    echo "User must specify start or stop as third argument"
+    exit 1
+fi
+
+if [ "$3" != "start" ] && [ "$3" != "stop" ]
+then
+    echo "User must specify start or stop as third argument"
+    exit 1
+fi
+
+# source the hive-env.sh file
+source $1/hive-env.sh
+myhostname=`hostname`
+
+if [ "$3" == "start" ]
+then
+    if [ -f "${HIVE_PID_DIR}/hive-metastore.pid" ]
+    then
+        echo "Hive metastore pid file found, it should already be running"
+    else
+        if [ "${HIVE_METASTORE_DIR}X" != "X" ]
+        then
+            hivemetastoredir=${HIVE_METASTORE_DIR}
+        else
+            hivemetastoredir="${hivelocalscratchdirpath}/hive/database/postgres/data"
+        fi
+       
+        $POSTGRES_HOME/bin/pg_isready -h localhost -p $PGPORT &> /dev/null &
+        postgrescheck=$?
+
+        if [[ $postgrescheck != 0 ]] 
+        then
+          echo "pg_isready reports: ${postgrescheck} -- Postgres not ready on ${PGPORT}.  Exiting Hive."
+          exit 1
+        fi
+        # check for the metastore warehouse directory, if not present create it.
+        if [ ! -d $HIVE_METASTORE_DIR ]
+        then
+            echo "Metastore warehouse directory ( ${HIVE_METASTORE_DIR} ) not found, creating directory now."
+            mkdir -p $HIVE_METASTORE_DIR
+        fi
+
+        echo "Starting Hive Metastore on ${myhostname}"
+
+        nohup $2/bin/hive --service metastore &> /dev/null &
+        scriptpid=$!
+        echo $scriptpid > ${HIVE_PID_DIR}/hive-metastore.pid
+    
+        sleep 10
+        export PATH=${PATH}:$2/bin
+        echo "Checking Hive Metastore settings"
+
+        # If the current node is different from the node where metastore is running the JDBC connector throws errors 
```

Comment:
how could the current node be different than the metastore node?  If it were different, is this actually an error?  Perhaps its a subtlety in Hive I don't understand.

Reply:
When the metastore/database is created in Hive it uses the hostname as part of the location.  During testing we encountered situations where it was created on one node and after the stack was brought down and back up on a different node Hive could not find the database.
The $HIVE_HOME/bin/metatool is used to correct that situation if/when it is encountered.

Reply:
Ohhh, you mean like the nodes are different between different runs on the cluster?  Like the first time hive ran on ```foo[0-10]``` and the second time it ran between ```foo[4-14]```?

That's fine then.  Perhaps adjust comments a bit to note what you said above.  The comments were just a tad confusing.